src/map/guild.c

```diff
@@ -217,10 +217,63 @@ static bool guild_read_castledb_libconfig_sub(struct config_setting_t *it, int i
 	}
 	safestrncpy(gc->castle_event, name, sizeof(gc->castle_event));
 
+	if (itemdb->lookup_const(it, "SeigeType", &i32) && (i32 >= SEIGE_TYPE_MAX || i32 < 0)) {
+		ShowWarning("guild_read_castledb_libconfig_sub: Invalid SeigeType in \"%s\", entry #%d, defaulting to SEIGE_TYPE_FE.\n", source, idx);
+		gc->seige_type = SEIGE_TYPE_FE;
+	} else {
+		gc->seige_type = i32;
+	}
+
+	libconfig->setting_lookup_bool_real(it, "EnableClientWarp", &gc->enable_client_warp);
+	if (gc->enable_client_warp == true) {
+		struct config_setting_t *wd = libconfig->setting_get_member(it, "ClientWarp");
+		guild->read_castledb_libconfig_sub_warp(wd, source, gc);
+	}
 	idb_put(guild->castle_db, gc->castle_id, gc);
 	return true;
 }
 
+static bool guild_read_castledb_libconfig_sub_warp(struct config_setting_t *wd, const char *source, struct guild_castle *gc)
+{
+	nullpo_retr(false, wd);
+	nullpo_retr(false, gc);
+	nullpo_retr(false, source);
+
+	int64 i64 = 0;
+	struct config_setting_t *it = libconfig->setting_get_member(wd, "Position");
+	if (config_setting_is_list(it)) {
+		int m = map->mapindex2mapid(gc->mapindex);
+
+		gc->client_warp.x = libconfig->setting_get_int_elem(it, 0);
+		gc->client_warp.y = libconfig->setting_get_int_elem(it, 1);
+		if (gc->client_warp.x < 0 || gc->client_warp.x >= map->list[m].xs || gc->client_warp.y < 0 || gc->client_warp.y >= map->list[m].ys) {
+			ShowWarning("guild_read_castledb_libconfig_sub_warp: Invalid Position in \"%s\", for castle (%d).\n", source, gc->castle_id);
+			return false;
+		}
+	} else {
+		ShowWarning("guild_read_castledb_libconfig_sub_warp: Invalid format for Position in \"%s\", for castle (%d).\n", source, gc->castle_id);
+		return false;
+	}
+
+	if (libconfig->setting_lookup_int64(wd, "ZenyCost", &i64)) {
+		if (i64 > MAX_ZENY) {
+			ShowWarning("guild_read_castledb_libconfig_sub_warp: ZenyCost is too big in \"%s\", for castle (%d), capping to MAX_ZENY.\n", source, gc->castle_id);
+			gc->client_warp.zeny = MAX_ZENY;
+		} else {
+			gc->client_warp.zeny = (int)i64;
```

Comment:
```suggestion
			gc->client_warp.zeny = cap_value((int)i64, 0, MAX_ZENY);
```
prevent negative value? and `ELSE` can be removed.

Reply:
I like having the warning for bad input case

Reply:
kept the showwarning, but just extra `cap_value` to avoid negative value, or add another checking for negative value?
Something like...
```
if (gc->client_warp.zeny > MAX_ZENY) {
    ShowWarning("guild_read_castledb_libconfig_sub_warp: ZenyCost is too big in \"%s\", for castle (%d), capping to MAX_ZENY.\n", source, gc->castle_id);
}
gc->client_warp.zeny = cap_value((int)i64, 0, MAX_ZENY);
```